spec/requests/file_spec.rb

```diff
@@ -2,17 +2,45 @@
 
 require 'rails_helper'
 
-RSpec.describe "Authentication for File requests", type: :request do
-  describe 'OPTIONS options' do
-    before do
-      allow(Purl).to receive(:public_xml).and_return('<publicObject />')
-    end
+RSpec.describe "File requests", type: :request do
+  before do
+    allow(Purl).to receive(:public_xml).and_return('<publicObject />')
+  end
 
+  describe 'OPTIONS options' do
     it 'permits Range headers for all origins' do
       options '/file/xf680rd3068/xf680rd3068_1.jp2'
       expect(response).to be_successful
       expect(response.headers['Access-Control-Allow-Origin']).to eq '*'
       expect(response.headers['Access-Control-Allow-Headers']).to include 'Range'
     end
   end
+
+  describe 'GET file with slashes in filename' do
+    let(:stacks_file) { StacksFile.new(id: 'xf680rd3068', file_name: 'path/to/xf680rd3068_1.jp2') }
+    let(:world_rights) do
+      <<-EOF
+        <publicObject>
+          <rightsMetadata>
+            <access type="read">
+              <machine>
+                <world/>
+              </machine>
+            </access>
+          </rightsMetadata>
+        </publicObject>
+      EOF
+    end
+
+    before do
+      allow(File).to receive(:world_readable?).and_return(nil)
```

Comment:
Why are we returning files that aren't readable? Is this stub necessary?

Reply:
@jcoyne Good eye! It was copypasta from the file auth request spec. Removed it and all is well.